src/app/portlets/dot-edit-page/layout/dot-additional-actions/dot-legacy-additional-actions-iframe/dot-legacy-additional-actions-iframe.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { DotMenuService } from '../../../../../api/services/dot-menu.service';
+
+@Component({
+    selector: 'dot-legacy-addtional-actions',
+    templateUrl: './dot-legacy-additional-actions-iframe.component.html'
+})
+export class DotLegacyAdditionalActionsComponent implements OnInit {
+
+    additionalPropertiesURL: string;
+
+    constructor(private route: ActivatedRoute, private dotMenuService: DotMenuService) {
+
+    }
+
+    ngOnInit(): void {
+        this.route.params.subscribe(params => this.setAdditionalPropertiesURL(params.id, params.tabName));
```

Comment:
You are doing 2 subscribe to get the information you need, for this cases you can use the ```.switchMap``` operator of RXJS, take a look here: https://youtu.be/65Us8NwmYf4?t=3m57s and I think you can use async pipe too.